### ISSUE TITLE
Test about sharding-scaling

### ISSUE THREAD

**Issue by ssxlulu** [avalon566/sharding-scaling#27]
I am doing a simple test about the sharding-scaling and I will write a test document after the test.

**Comment by avalon566** [comment]
Great!
Here is some test scripts for shrading-scaling [sharding-scaling-test](https://github.com/avalon566/sharding-scaling-test), I think is useful for you.

**Comment by ssxlulu** [comment]
@avalon566 I am being testing.And I found the speed is too slow.I migrate one database with one table with 10,000 thousands data items to another database. And only 11w data items have been migrated after 1 hour.
The table has no primary key and no indexes.

**Comment by avalon566** [comment]
Tuning performance work will start after the feature and code is stable and no serious bugs.

**Comment by avalon566** [comment]
@ssxlulu I've fixed a bug, you can test the speed again.

**Comment by ssxlulu** [comment]
@avalon566 I found it is too slow to use...
It takes me about 3d 17h to migrate about 960w data items, and these data's size only less than 4GB.

**Comment by avalon566** [comment]
Do you pull the newest code? I test with ssd disk, 100w data cost 10 minute.

**Comment by ssxlulu** [comment]
> Do you pull the newest code? I test with ssd disk, 100w data cost 10 minute.

I use the Last Saturday's code, and the code may not change a lot after that?

**Comment by avalon566** [comment]
Yes, you can checkout this commit 13b1570, and test it.

**Comment by ssxlulu** [comment]
Following is the test figure.

![image](https://user-images.githubusercontent.com/14773179/67859286-2ea78300-fb56-11e9-9ddd-5886f422970e.png)

And the following is the statistics of HOST B:

![image](https://user-images.githubusercontent.com/14773179/67860059-f6a13f80-fb57-11e9-96ed-119e19d2c15a.png)

![image](https://user-images.githubusercontent.com/14773179/67860099-0751b580-fb58-11e9-9a36-234f72246cc8.png)

![image](https://user-images.githubusercontent.com/14773179/67860153-1f293980-fb58-11e9-8bab-a0b44d696ddd.png)

![image](https://user-images.githubusercontent.com/14773179/67860299-716a5a80-fb58-11e9-92b7-e4d2b9071357.png)

In conclusion, The disk I/O is idle, and the network traffic is idle too. 
I want to increse the batch size of data items it migrates each time, but i can not find the parameter.We may make it can be configured.

**Comment by avalon566** [comment]
Great!
I will finish this feature.

**Comment by avalon566** [comment]
And Welcome to contribute this feature.
